de.dlr.sc.virsat.model.extension.budget.cost.ui/src/de/dlr/sc/virsat/model/extension/budget/cost/ui/snippet/UiSnippetTableCostEquipment.java

```diff
@@ -23,4 +38,26 @@
  * 
  */
 public class UiSnippetTableCostEquipment extends AUiSnippetTableCostEquipment {
+	@Override
+	protected void createButtons(FormToolkit toolkit, EditingDomain editingDomain, Composite sectionBody) {
+		
+		super.createButtons(toolkit, editingDomain, sectionBody);
+		Button addCurrencyButton = toolkit.createButton(sectionBody, "Add Currencies to Model", SWT.PUSH);
+
+		addCurrencyButton.addSelectionListener(new SelectionListener() {
+
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				
+				VirSatTransactionalEditingDomain virSatEd = VirSatEditingDomainRegistry.INSTANCE.getEd(model);
```

Comment:
Should moved to model code.

Reply:
Right would be an improvement to move this code to a separate helper class in the model code plugin and just call it here.

Reply:
Code outsourced in its own method (EuroUnitCreator).